components/merkledb/src/values.rs

```diff
@@ -109,6 +109,45 @@ macro_rules! impl_binary_value_scalar {
             }
         }
 
+        impl_object_hash_for_binary_value! { $type }
+    };
+    ($type:tt, $write_32:ident, $read_32:ident, $len_32:expr, $write_64:ident, $read_64:ident, $type_64:tt, $len_64:expr) => {
+        #[cfg(target_pointer_width = "32")]
```

Comment:
I don't think that platform-specific encoding is a good idea for `MerkleDB` traits, since the data layout should be the same for every node in the network. With such an approach, it can cause the hash divergence on different nodes (e.g. if one node runs on a 32-bit PC, and one on a 64-bit PC, and some merkelized index calculates hash from the `BinaryValue`-encoded data).

Reply:
Thank you for your point of view.

I agree with you. I think it might be better to store platform-specific values as a 32-bit integer. But in this case, we need to solve the 32-bit variable overflow problem. This can happen if the buffer contains a value greater than 4 GB. But I think this is such an unlikely case to store huge unit values in the blockchain so that we can do this.

In addition, buffers > 2^32 bytes are not supported by 32-bit systems.

Reply:
We can store platform-specific values in `u64`, which will fit both 32-bit and 64-bit pointers :)

Reply:
But in this case we have to store useless zeroes in rest 4 bytes for each size value.

Reply:
I think that it's not a big problem: 
1. There are not so many 32-bit machines nowadays.
2. It's not a pretty common scenario to use `usize` / `isize` as a value to store. If one will be totally sure that value will never exceed 32-bit limit, they will be able to use `u32` / `i32` instead.